### DISCUSSION TITLE
Is there a way to cache pyenv versions between runs on CI servers?

### DISCUSSION THREAD

**Discussion by carlosjgp** [pyenv/pyenv#3083] [Q&A]
In particular, I'm using CircleCI.

I feel like our pipelines are doing extra work each time downloading the same ZIP file (could cache this) and installing the version (or cache this)

I've been searching on CCI discuss and on this repository code, issues and wiki but couldn't find anything

**Comment by native-api** [comment]
You can cache the `$PYENV_ROOT/versions` directory or the required subdirectories.

But first, check if the CI itself provides a way to install Python of the corresponding version. They probably do such caching themselves as part of that (e.g. TravisCI did).

**Reply by carlosjgp** [reply]
I tried that but I end up with a long list of symlink errors... that's why I was wondering if I could cache the Zip file download instead of the installation of the shim

I worry that restoring the `versions` folder would cause errors later 

And yeah, CCI provides their own installation of pyenv
